File: concepts/SOLID/Java/3-LSP/after/GlobomanticsHR/src/main/java/hr/personnel/Subcontractor.java
package hr.personnel;

public class Subcontractor{

    private final int monthlyIncome;
    private int nbHoursPerWeek;
    private String email;
    private String name;

    public Subcontractor(String name, String email, int monthlyIncome, int nbHoursPerWeek) {
        this.name = name;
        this.email = email;
        this.monthlyIncome = monthlyIncome;
        this.nbHoursPerWeek = nbHoursPerWeek;
    }

    public boolean approveSLA(ServiceLicenseAgreement sla) {
        /*
        Business logic for approving a
        service license agreement for a
        for a subcontractor
         */
        boolean result = false;
        if (sla.getMinUptimePercent() >= 98
                && sla.getProblemResolutionTimeDays() <= 2) {
            result=  true;
        }

        System.out.println("SLA approval for " + this.name + ": " + result);
        return result;
    }
}

File: concepts/SOLID/PHP/4-ISP/b/src/personnel/Subcontractor.php
<?php 
namespace app\personnel;

use RuntimeException;

class Subcontractor  
{  
  private $name;
  private $email;
  private $monthlyIncome;
  private $nbHoursPerWeek;
  
  public function __construct(string $name, string $email, int $monthlyIncome, int $nbHoursPerWeek)
  {
    $this->name   = $name;
    $this->email  = $email;
    $this->monthlyIncome = $monthlyIncome;
    $this->nbHoursPerWeek = $nbHoursPerWeek;
  }

  public function approveSLA(ServiceLicenseAgreement $sla)
  {
      $result = false;
      if($sla->getMinUptimePercent() >= 98 && $sla->getProblemResolutionTimeDays() <= 2){
        $result = true;
      }
      echo "SLA approval for {$this->name}: {$result} \n";
      return $result;
  }
}
